bluemira/utilities/tools.py

```diff
@@ -101,6 +101,96 @@ def json_writer(
     return None
 
 
+# =====================================================
+# csv writer utilities
+# =====================================================
+def write_csv(
+    data: np.ndarray,
+    base_name: str,
+    col_names: list[str],
+    metadata: str | None = "",
+    ext: str = ".csv",
+    comment_char: str = "#",
+):
+    """
+    Write data in comma-separated value format.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array of data to be written to csv file. Will raise an error if the
+        dimensionality of the data is not two
+    base_name : str
+        Name of file to write to, minus the extension.
+    col_names : list(str)
+        List of strings for column headings for each data field provided.
+    metadata: str
+        Optional argument for metadata to be written as a header.
+    ext : str
+        Optional argument for file extension, defaults to ".csv".
+    comment_char : str
+        Optional argument to specify character(s) to prepend to metadata lines
+        as a comment character (defaults to "#").
+
+    """
+    # Fetch number of cols
+    shape = data.shape
+    # replacing `2` with a constant variable to avoid PLR2004
+    min_shape_length = 2
+    n_cols = 1 if len(shape) < min_shape_length else shape[1]
```

Comment:
In this instance we dont need to
```suggestion
    n_cols = 1 if len(shape) <  2 else shape[1]  # noqa: PLR2004
```